packages/data-schema/src/SchemaProcessor.ts

```diff
@@ -1450,8 +1568,27 @@ const schemaPreprocessor = (
           databaseType,
           getRefType,
         );
+        inputTypes.push(...operationInputTypes);
+        gqlModels.push(...operationReturnTypes);
 
-        topLevelTypes.push(...implicitTypes);
+        implicitTypes.forEach(([name, type]) => {
+          if (isEnumType(type)) {
+            const enumType = `enum ${name} {\n  ${type.values.join('\n  ')}\n}`;
+            enumTypes.add(enumType);
+          } else {
+            const isInput = !operationReturnTypes.some((rt) =>
+              rt.includes(name),
+            );
+            const generatedType = generateInputTypes(
+              [[name, type]],
+              isInput,
+              getRefType,
+            )[0];
```

Comment:
Broader than this section, but I've been stuck in this section of code trying to parse out whats happening here for a few minutes. This block in particular is especially difficult to grok.

The SchemaProcessor has been a growing pile of ver pragmatic solutions for a while, that causes it to be hard to work/reason across. I don't think we should refactor it with this change, but this change has me wanting to prioritize a refactor.

In the meantime are there opportunities to make a pass across these changes aiming at adding clarity in structure, ie making it easier to understand for a reviewer or the next dev who needs to make related changes?